src/shared/infra/email/providers/email-provider.ts

```diff
@@ -0,0 +1,3 @@
+export interface EmailProvider {
```

Comment:
Man, sei que já aprovei teu PR. Mas dando uma revisada de novo aqui. Acho seria interessante mudar o nome dessa interface e da pasta para adapters, ai os usecase/service utilizariam essa interface, e nos factory iriamos injetar a implementação dela

```suggestion
export interface EmailAdapter {
```